### ISSUE TITLE
Library view is not recognized as UI element by Windows Narrator and other screen readers

### ISSUE THREAD

**Issue by mutterback** [readium/readium-desktop#214]
The impact of this issue is that a user cannot access the list of EPUB titles in their library using a screen reader, and therefore cannot open a book using a screen reader. We've reproduced this issue with Windows Narrator and JAWS. 

(Note - we will update the issue with a list of the UI elements Narrator _can_ access according to Windows UI Accessibility Checker.)

**Comment by kevinmainairungu** [comment]
Below is the list of UI elements that the screen reader can be able to read.  
![screenshot 65](https://user-images.githubusercontent.com/23236275/53471558-243c6300-3a76-11e9-9817-2ff005ed45e2.png)

**Comment by mutterback** [comment]
@kevinmainairungu can you upload the latest UI Accessibility checker report? As of eKitabu's latest compile, Narrator can't see any menu items in the app.

**Comment by brettins** [comment]
Not sure how relevant this is, but it's possible that there's a fundamental difference in the systems being used with Electron & Navigator. 

https://github.com/electron/electron/issues/11867

**Comment by danielweck** [comment]
Yes, Narrator's reliance on the UIAutomation protocol (instead of MSAA or IAccessibleX) has been problematic in several software projects I have been involved in. Thankfully NVDA and JAWS typically "hand-shake" better with Chromium.

Also note that in that light, the move to Chromium in Edge has been concerning but on the plus side this means Microsoft will likely allocate resources to help address issues with assistive technology.

More info:
https://docs.microsoft.com/en-us/windows/desktop/winauto/microsoft-active-accessibility-and-ui-automation-compared#choosing-microsoft-active-accessibility-ui-automation-or-iaccessibleex

**Comment by danielweck** [comment]
Note that NVDA "hand-shakes" relatively nicely with Thorium now (develop branch). There is plenty more to do to make the entire user interface usable with a screen reader, but at least with have a decent start.

**Comment by mutterback** [comment]
Quick update on this from Microsoft:

Support for IAccessible2 – the same accessibility API that is implemented in Electron – was introduced in Windows 1903 aka “19H1” aka “May update”.

This is broadly released now, to install see:

https://www.microsoft.com/en-ca/software-download/windows10

also see

https://support.microsoft.com/en-us/help/4028685/windows-10-get-the-update

**Comment by danielweck** [comment]
Yes, this is encouraging!

UI Automation framework IAccessible2 Support
From Windows 10 May 2019 update version 1903, the UI Automation framework will support the translation of IAccessible2 info from Google Chrome, which supports IAccessible2 instead of than UI Automation. This feature makes UI Automation clients capable of getting the IAccessible2 information using the UI Automation client interfaces.

See "Version 1903 Accessibility improvements"
https://thewincentral.com/windows-10-19h1-version-1903-all-changes-new-features-so-far-categorized-changelog/#Version_1903_Accessibility_improvements

**Comment by llemeurfr** [comment]
since we loaded the latest Windows 10 v1903, Narrator works correctly (proper focus on the library view on launch etc.). I propose to close this issue and open discrete issues where problems occur.

**Comment by mutterback** [comment]
Agreed, closing this issue.
